packages/rocketchat-ui-login/client/login/form.js

```diff
@@ -0,0 +1,290 @@
+				if (RocketChat.settings.get('LDAP_Enable')) {
+					loginMethod = 'loginWithLDAP';
+				}
+				if (RocketChat.settings.get('CROWD_Enable')) {
+					loginMethod = 'loginWithCrowd';
+				}
+				return Meteor[loginMethod](s.trim(formData.emailOrUsername), formData.pass, function(error) {
+					const user = Meteor.user();
+					RocketChat.Button.reset(button);
+					if (error != null) {
+						if (error.error === 'no-valid-email') {
+							instance.state.set('email-verification');
+						} else {
+							toastr.error(t('User_not_found_or_incorrect_password'));
+						}
+						return;
+					}
+					if (user && user.language) {
+						localStorage.setItem('userLanguage', user.language);
+						return setLanguage(Meteor.user().language);
+					}
+				});
+			}
+		}
+	},
+	'click .register'() {
+		Template.instance().state.set('register');
+		return RocketChat.callbacks.run('loginPageStateChange', Template.instance().state.get());
+	},
+	'click .back-to-login'() {
+		Template.instance().state.set('login');
+		return RocketChat.callbacks.run('loginPageStateChange', Template.instance().state.get());
+	},
+	'click .forgot-password'() {
+		Template.instance().state.set('forgot-password');
+		return RocketChat.callbacks.run('loginPageStateChange', Template.instance().state.get());
+	},
+	'click .one-passsword'() {
+		if (OnePassword && OnePassword.findLoginForUrl) {
+			return;
+		}
+		const succesCallback = function(credentials) {
+			$('input[name=emailOrUsername]').val(credentials.username);
+			return $('input[name=pass]').val(credentials.password);
+		};
+		const errorCallback = function() {
+			return console.log('OnePassword errorCallback', arguments);
+		};
+		return OnePassword.findLoginForUrl(succesCallback, errorCallback, Meteor.absoluteUrl());
+	}
+});
+
+Template.loginForm.onCreated(function() {
+	const instance = this;
+	this.customFields = new ReactiveVar;
+	Tracker.autorun(() => {
+		const Accounts_CustomFields = RocketChat.settings.get('Accounts_CustomFields');
+		if (typeof Accounts_CustomFields === 'string' && Accounts_CustomFields.trim() !== '') {
+			try {
+				return this.customFields.set(JSON.parse(RocketChat.settings.get('Accounts_CustomFields')));
+			} catch (error1) {
+				return console.error('Invalid JSON for Accounts_CustomFields');
+			}
+		} else {
+			return this.customFields.set(null);
+		}
+	});
+	if (Meteor.settings['public'].sandstorm) {
+		this.state = new ReactiveVar('sandstorm');
+	} else if (Session.get('loginDefaultState')) {
+		this.state = new ReactiveVar(Session.get('loginDefaultState'));
+	} else {
+		this.state = new ReactiveVar('login');
+	}
+	this.validSecretURL = new ReactiveVar(false);
+	const validateCustomFields = function(formObj, validationObj) {
+		const customFields = instance.customFields.get();
+		if (!customFields) {
+			return;
+		}
+
+		for (const field in formObj) {
+			if (formObj.hasOwnProperty(field)) {
+				const value = formObj[field];
+				if (customFields[field]) {
```

Comment:
I guess you inverted this condition, the correct is `if (customFields[field] == null) {`